### PR TITLE
Fix weight

### LINKED ISSUE #48724: The weight() function in item.cpp do not make use of all of its arguments
# Is your feature request related to a problem? Please describe.
The function that return the mass of an object is only partially implemented.
https://github.com/CleverRaven/Cataclysm-DDA/blob/b991a1bab26d0fb8a6f9068ab4e9c9639ab76e26/src/item.h#L569
The parameter `include_contents` is not used.

# Describe the solution you'd like
The function could use the `include_contents` parameter. As a guide, the function returning the volume of an object seems fully implemented and has similar parameters : 
https://github.com/CleverRaven/Cataclysm-DDA/blob/1ae66c97a0db1b31a9444190f59d5b58149327d1/src/item.cpp#L5327

# Describe alternatives you've considered
While I think we can use the `itype` of an container item to get its empty mass, I suppose it won't work if the said item has mods attached. Also, we may want the mass of an item, with its battery, but not its others content if it is a liquid container/sheath.  ( I don't know if the damage level change the mass, maybe it is also a case where it is a problem )

# Additional context
The function is called weight instead of mass but this is out of scope.
I've rammed existing issues, trying not to duplicate one, but as there is a lot of opened issues concerning weight, I may have missed it/them.

### PR DESCRIPTION
**Pull request by WelpThatWorked**
#### Summary
Infrastructure "Improve weight() implementation"

Update weight to use include_contents parameter, rather than always including contents weight. The behavior of existing calls should not be affected by this change.
#### Purpose of change
Fix #48724 

#### Describe the solution
Use include_contents parameter rather than always including content weight. Remove separate adding of gunmod and magazine weight, as they are covered by the item_weight_modifier, and thus caused double counting of those items.

#### Testing
Placing and removing items from containers, Equipped weapon, added modification, fired weapon until empty, unloaded weapon. Weight was reported correctly after each step. 

#### Additional context

### CODE REVIEW

**Comment by jbytheway** [comment]
This is something it should be easy to write unit tests for.  If we had those, they perhaps would have caught this double-counting of gunmods and ammo already.  Would you be able to add some?  Looks like there's one weight-related test in `tests/item_contents_test.cpp` already, but evidently we need more.

**Comment by WelpThatWorked** [comment]
Just to clarify, the double-counting thing was something I did by mistake because I didn't really understand the item structure, It did not exist before my first commit. I'm haven't written a unit test before, but I could if I knew what to test for. What would I add to the test?

**Comment by jbytheway** [comment]
In this case, just create a couple of `item` objects, with and without contents, and verify that the weights are the expected values for `include_contents` being `true` or `false`.  For example, see [this test](https://github.com/CleverRaven/Cataclysm-DDA/blob/e4d05ed8582bba818a4a96502d2fa51af9ceaab9/tests/item_contents_test.cpp#L54-L55), which is the only current weight test we have, I think.

**Comment by WelpThatWorked** [comment]
I added on to the existing test, does it seem right?

**Comment by jbytheway** [comment]
> I added on to the existing test, does it seem right?

Yes, that looks good to me :+1:.

File: src/item.cpp [hunk 1]
```diff
@@ -5131,6 +5131,7 @@ units::mass item::weight( bool, bool integral ) const
         ret *= 0.75;
     }
 
+
```
Comment (kevingranade) [inline src/item.cpp:5134]:
```suggestion
```

**Pull request merged**